src/region/RegionAnnotations.tsx

```diff
@@ -41,22 +35,22 @@ export default class RegionAnnotations extends React.PureComponent<Props, State>
 
     state: State = {};
 
+    componentDidUpdate(_prevProps: Props, prevState: State): void {
+        const { setReferenceShape, staged } = this.props;
+        const { rectRef } = this.state;
+        const { rectRef: prevRectRef } = prevState;
+
+        if (prevRectRef !== rectRef && rectRef && staged) {
```

Comment:
Is `&& staged` needed here or can we dispatch the action any time rectRef changes and is defined?

Reply:
Good call, I'll remove staged